config/initializers/multifield.rb

```diff
@@ -54,6 +54,7 @@ def render
     options['value'] = options.fetch('value'){ value_before_type_cast(object) }
     options['value'] &&= ERB::Util.html_escape(options['value'])
     add_default_name_and_id(options)
+    options['data-name'] = options['name']
```

Comment:
might be worth a comment explaining why data-name is being used instead of `options['name']`